### ISSUE TITLE
Chrome fails to launch

### ISSUE THREAD

**Issue by Manouchehri** [Microsoft/BashOnWindows#648]
* A brief description

Chrome fails to launch.
* Your Windows build number

10.0.14385
* Commands required to reproduce the error
```
/usr/bin/google-chrome
```
* Strace of the failing command

https://gist.githubusercontent.com/Manouchehri/da19ce4b9874c0a893210fa509fc1a33/raw/5fea50f4e81233e1cbe65ccdd92a295352ae9197/strace-chrome-wsl-fail.txt

* Required packages and commands to install

```bash
wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | sudo apt-key add -
sudo apt-get update
sudo apt-get install google-chrome-unstable
```

**Comment by imkow** [comment]
As BashOnWindows is intended to development purposes, rather than normal usage scenarios with GUIs,
I think this isn't a problem. Except it's not about GUIs.

**Comment by Manouchehri** [comment]
@imkow GUIs definitely work. See #637.

![Hopper](https://cloud.githubusercontent.com/assets/7232674/16747715/534a5ad6-478e-11e6-8117-ff29d8d8fdde.png)

**Comment by Gabrielcarvfer** [comment]
The problem is related to sockets. As Chromium relies on sockets for IPC in Linux and BSD, you can expect other chromium based stuff to fail (Chrome/Opera/Spotify/...).

**Comment by aseering** [comment]
For what it's worth, there are a number of existing bugs filed in this bug tracker against WSL sockets.  (The WSL folks have been working on fixing them.)  If you have time to look through them, it would be interesting to know if this is a known or new underlying issue.

**Comment by bitcrazed** [comment]
@Manouchehri While we greatly appreciate your taking the time to stretch Bash/WSL and investigate this issue, we are being quite explicit in that we do not support GUI/desktop apps; we are only supporting command-line applications at this time.

That several GUI apps do run (to a greater or lesser degree), is purely a side-effect of the team improving our syscall implementation over time.

We'll be sure to (very) publicly announce if this position changes, but expect that we'll only be supporting command-line scenarios and tools for the foreseeable future.

**Comment by Manouchehri** [comment]
Well, I was trying to pinpoint the issue with sockets rather than the GUI..

I'll write an example that doesn't use a GUI since apparently that's too distracting.

**Comment by Gabrielcarvfer** [comment]
Tried some stuff using socketpair, but it worked.

The only thing for sure is that the error happens in a check at line 40 from https://github.com/scheib/chromium/blob/master/content/browser/renderer_host/render_sandbox_host_linux.cc

**Comment by Gabrielcarvfer** [comment]
Seems related to #546 and #514.

**Comment by therealkenc** [comment]
You can work around the render_sandbox_host_linux.cc easy enough. There are lots of missing pieces before Chrome runs. The issue as raised by @Manouchehri is not well formed and I'm glad this has been marked bydesign.

Anyway, I hit the following for starters:

1. `dir_reader_linux.h`: WSL doesn't like something about casting to` linux_dirent*` (I think; I hit this before I started taking notes). Can work-around by using `DirReaderFallback`.
2. `render_sandbox_host_linux.cc`: Half-close on socketpair(AF_UNIX, SOCK_SEQPACKET, ...). #514. This can be worked around easily as it happens by just not calling `shutdown()`.
3. `thread_helpers.cc (44)`: Check failed: 3UL <= task_stat.st_nlink (3 vs. 1). This module queries `proc/self/task/` and doesn't like what it sees. Code in question looks like:

    ```
        // At least "..", "." and the current thread should be present.
        CHECK_LE(3UL, task_stat.st_nlink);    // <--- this fails on WSL, commented out check.
        // Counting threads via /proc/self/task could be racy. For the purpose of`
        // determining if the current proces is monothreaded it works: if at any
        // time it becomes monothreaded, it'll stay so.`
        return task_stat.st_nlink == 3;       // <--- wrong return value on WSL (hacked to <= 3)
    ```
4. `stack_trace_posix.cc` (584): Failed to parse the contents of /proc/self/maps. I haven't looked into why.
5. `file_path_watcher_linux.cc`: general unhappiness due to lack of `inotify` in WSL #216. Replace with file_path_watcher_stub.cc
6.  The work around in #637 doesn't take because chrome wants the unix socket for dbus, period #376 Can probably be worked-around but I didn't bother, because...
7. `address_tracker_linux.cc(172)`: Could not bind NETLINK socket: Invalid argment. #264

I quit at the NETLINK issue. There are some others I haven't listed; notably chrome has it's own hand rolled version of `strace` built into the debug build that doesn't work for reasons.

**Comment by therealkenc** [comment]
I should add that the [Mozilla](https://www.mozilla.org/en-US/firefox/new/?utm_source=google&utm_medium=paidsearch&utm_campaign=Firefox-Brand-CA-EN-GGL-Exact&utm_term=firefox&gclid=CjwKEAjwtqe8BRCs-9DdpMOilBoSJAAyqWz_YVMY10yFMrpuDlHla6_bIyp-c0_QftplRTmcr0FKNBoCUL_w_wcB) people deserve a big kudos for writing sane portable posix code #481. Chrome has no business firing up a file watcher, having a whole IP address tracking subsytem, or poking around `/proc`. It's a **browser**, Google.

**Comment by Manouchehri** [comment]
I disagree that it's a poor question; opening this issue got you to clearly identify ~7 issues for the WSL team.

**Comment by therealkenc** [comment]
No, it didn't. Not even a little bit. Four of the seven issues are known to the WSL team. That Google dicks with Linux kernel internals in (1), (3), and (4) is on Google not Microsoft. Seeing what it would take to get Chrome working was a exercise of no value to the WSL team.

**Comment by Gabrielcarvfer** [comment]
Google guys don't agree with you. It's a whole OS inside a browser. xD

**Comment by SRGOM** [comment]
Can we keep bug requests sane please? Any second we waste of this team is a second lost from the resolution of #216 which is a major blocker for a lot of dev workflows.

**Comment by therealkenc** [comment]
And to be clear, _sane_ is defined as: a test case that demonstrates failing syscall or `/proc /dev /sys` interface that blocks use of "common Linux tools (sed, awk, etc.) and many Linux-first tools (Ruby, Python, etc.) to use their toolchain on Windows". 

Maybe if the WSL team added a `<blink>` tag to the [FAQ](https://msdn.microsoft.com/en-us/commandline/wsl/faq) would that help? If you don't know how to post a test case then continue using Ubuntu on a VM with samba shares like everyone else stuck in cross-platform hell.

**Comment by aseering** [comment]
Folks, this is getting a smidgeon heated...

First -- thanks @Manouchehri for reporting this!  It is a thing that WSL cannot currently do.  While it's not on the short-term road map, I for one do hope that WSL eventually reaches a level of completeness where it's able to run apps like Chrome.

It's not something that is expected to work in the Anniversary Release.  But as has been pointed out on other tickets, there are WSL users who (rightly or wrongly) expect browsers to work now.  This ticket documents what their actual status is.

@therealkenc -- your definition of "sane" matches what's expected to work for Windows Anniversary Edition.  But this bug tracker has a broader scope than that; it is currently described as encompassing future and long-term work as well.  If you would like to narrow its definition to match your description, please post a pull request modifying [README.md](https://github.com/Microsoft/BashOnWindows/blob/master/README.md).

**Comment by therealkenc** [comment]
Unfortunately yes, you are correct it would have been posted eventually -- because people don't read FAQs. Everyone who installed Chrome in April and saw it didn't work knew better than to open an issue. Except you. What information did you think you were conveying to the WSL team? That they didn't know Chrome doesn't run? If so, I am sure they appreciated your public service announcement.

That said, there is a legitimate matter regarding singling out "GUI apps" or "server apps". I know (and I'm sure @bitcrazed knows) that is a generalization. That an app has a GUI is kind of beside the point. Basically the entire Linux 2.6 (say) surface needs to be covered or WSL is going to cause patchkit hell for "Linux first" software; like in the bad days of SunOS, DG/UX, SCO, Tru64. I am sure the WSL team is aware of that too. But per @SRGOM, until they have covered all the _known_ surface that is missing, posting out of scope issues is not constructive or even informative.

**Comment by benhillis** [comment]
This is getting out of hand, I'm going to lock this thread and let people calm down.

**Comment by russalex** [comment]
Thanks @aseering and @benhillis.

We appreciate people pushing the limits to what WSL can do. This tracker has become one of the best places where people can show off and ask questions about areas that are not officially supported  (examples #637, #611, and #481). I know I get a kick out of reading these and do not want us to do anything to stifle the creativity.

For the issue, @Manouchehri, thank you for reporting it. If nothing else people who try and run Chrome can see other people's experience. We all know that Chrome is not a priority, but it is still good information to have. Also, for the record, thank you for the help with CONTRIBUTING.md.

The Windows Anniversary Update is pretty well locked down at this point. We have a [blog post](https://blogs.msdn.microsoft.com/wsl/2016/07/08/bash-on-ubuntu-on-windows-10-anniversary-update/) talking a bit about what comes next. Naturally our priority list will grow for the next release.  No spoilers on our conversations but I can promise that we're looking at this forum and the [User Voice](https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo/category/161892-bash). More information on that when we have something fun to share.

**Comment by Krzysztof-Cieslak** [comment]
Hey, 

Sorry for revive already closed issue, but haven't wanted to create another issue for basically same problem.

I'd like to just point out that this issue is connected not only with Chrome, but also with any other application using Chromium engine - including any Electron application.

This last group includes popular text editors such as Atom... and amazing MSFT product - VS Code. It would be really nice if you've reconsider fixing this issue to support development tools created by other teams in Microsoft. :)

Best regards,
Chris.

**Comment by iz0eyj** [comment]
Alot of programs fails on shutdown, please fix it (or waiting for the fix, simulate a correct return value)

**Comment by aseering** [comment]
@iz0eyj -- thanks for your comment!  It's very general, though...  Are you referring specifically to programs that use Chrome or the Chromium engine?

If so, could you please post more details?  If not, or if you're not sure, could you please open a new issue and fill out the issue template?

**Comment by iz0eyj** [comment]
@aseering is, in my case the problem is with Vivaldi Browser, which uses exactly like Chrome engine Blink (derived from Webkit), but probably also with Opera you will have the same error.

This is the end of his strace (shutdown at the second row):
socketpair(PF_LOCAL, SOCK_SEQPACKET, 0, [6, 7]) = 0
shutdown(6, SHUT_RD)                    = -1 EINVAL (Invalid argument)
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
tgkill(2512, 2512, SIGABRT)             = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=2512, si_uid=1000} ---
+++ killed by SIGABRT (core dumped) +++
Annullato (core dump creato)
federico@BESTIA:~/Scaricati$

**Comment by therealkenc** [comment]
Ben says #546 is better in 14915.

**Comment by iz0eyj** [comment]
@therealkenc I confirm that close semms OK with 14915, but I got a new issue (not socket related) starting Vivaldi Browser.
I opened the thread.

**Comment by therealkenc** [comment]
I've got a partial port of Chrome 49.0.2623.75 limping on 14915 (barely). It is too unstable to be usable at the moment, but if I make further progress I'll post patches and build instructions.

![chrome-wsl](https://cloud.githubusercontent.com/assets/7049700/18234138/cf6acb9a-72af-11e6-828e-3fd02942b9f2.png)

**Comment by mgiuffrida** [comment]
@therealkenc Please do post what you've managed to get so far! I'd be interested to see how far this could get and if there are any underlying bugs/assumptions in Chromium that would be feasible to fix, opinions about reading `/proc` files aside :-)

**Comment by therealkenc** [comment]
I'll be sure post progress if I make any. Right now I am blocked on:
```
futex(0x7fffda346fbc, FUTEX_CMP_REQUEUE_PRIVATE, 1, 2147483647, 0x7fffda346f90, 2) = -1 
   ENOSYS (Function not implemented)
```
...with the code in question being `pthread_cond_broadcast()` [here](https://cs.chromium.org/chromium/src/base/synchronization/condition_variable_posix.cc?q=ConditionVariable::Broadcast&sq=package:chromium&l=127&dr=CSs). This new problem is not quite the same as the `futex` problem in #1006, which is easy to work around. 

Which is not all that noteworthy in of itself; it is just where I happened to stop. I've been posting issues if/when I make the effort to do test cases the team can use.

**Comment by therealkenc** [comment]
I looked at this again last weekend and made a little more progress. My current patch against is 52.0.2743.82 [here](https://gist.github.com/therealkenc/b239f169773816566aff7d58eb08f775). The patch is enough to get chrome running without `--single-process` (ie with multiprocess), by working around #514 and #706. It still has too many problems to be usable, but it is just usable enough to send this post.

The `FUTEX_CMP_REQUEUE_PRIVATE` #1006 problem I mentioned above was a false positive of sorts. It's still a missing WSL feature, but glibc pthreads seems to fallback awkwardly (I'd hate to say gracefully) to `FUTEX_WAKE_PRIVATE`, which is supported.

This patch is also enough to get libchromiumcontent 1.3.8 to launch, but (sadly) not enough for VS Code to fire up yet.

![wsl-electron](https://cloud.githubusercontent.com/assets/7049700/19989110/7d5f5a26-a1e1-11e6-8e06-603c053ad29b.png)

**Comment by aseering** [comment]
Ooh, nice progress!

**Comment by mgiuffrida** [comment]
FWIW the `/proc/sys/kernel/shmmax` read was removed from chrome [back in June](https://codereview.chromium.org/2054593002.

**Comment by therealkenc** [comment]
I noticed that, which is part of the reason I didn't submit a bug. Unfortunately the change only landed in Chrome 53 (stable) in September, and VS Code and Atom are still on 52. 

Silliest magic number in the history of computing. `18446744073692774399`. That's right: `eighteen quintillion, four hundred forty-six quadrillion, seven hundred forty-four trillion, seventy-three billion, six hundred ninety-two million, seven hundred seventy-four thousand, three hundred ninety-nine`. Twenty significant digits, and it isn't even divisible by a page size.

Five min fix if the team wants to pick it up. Has to be done eventually anyway. But make `shmmax` return decimal value of `0x2000000000000000` so you don't have to feel quite as sad about it.

**Comment by tobireif** [comment]
Could I install Chrome on Windows (independent of Bash-On-Windows), and then start that Chrome from the Bash shell? (In oder to open eg .html files from within the shell.)

Also, I have this in my .vimrc, and would hope it will work when I switch to Windows / Bash-On-Windows:
````
" Go to URL ("Go to Link"):
if has("macunix")
  :map gl :!open -a "Google Chrome" '<cWORD>'<CR>
elseif has("win32") || has("win64")
  :map gl :!chrome "<C-R><C-A>"<CR>
else
  :map gl :!chrome "<cWORD>" &<CR>
endif
````

**Comment by tobireif** [comment]
P.S. I want to start GUI apps as Windows apps (installed on Windows), not as Linux apps.

**Comment by therealkenc** [comment]
Sure, the following from the command prompt will get you right back here. You need to be on an insider build 14951 or better which has the Win32 interop stuff.
```
"/mnt/c/Program Files (x86)/Google/Chrome/Application/chrome.exe" \
    --profile-directory="Default" https://github.com/Microsoft/BashOnWindows/issues/648
```

**Comment by tobireif** [comment]
@therealkenc Thanks!

**Comment by therealkenc** [comment]
Got a bit further this weekend. Chrome 53.0.2785.143 now runs _somewhat_ stable on 15042 with this small [patch](https://gist.github.com/therealkenc/b82a5cc4720b559fd751296d60f799b4). Stable enough to play youtube videos anyway, which was a non-starter in November. It still does an 'awe snap' if you try hard enough. As you can see, the main blocker for chrome running out of the box remains [udev](https://en.wikipedia.org/wiki/Udev).

The patch is a lot smaller because #1326 was squashed (thanks Sunil). I can't explain precisely what caused the stability improvement. The patch doesn't work around anything _new_, and the previous one worked around #1326 already. Could any number of random WSL fixes since November. Could be changes in chromium since 52.0.2743.82. 

VSCode, unfortunately, remains obstinate. This is frustrating, because "it should be working now". Sigh. I'll bang on it some more.

ob-screencap:

![chrome-wsl-feb27](https://cloud.githubusercontent.com/assets/7049700/23355479/bc7bb2dc-fc8a-11e6-9c18-11c8d8d6aa5b.png)

**Comment by therealkenc** [comment]
I have written a stub library for `libudev.so.1` that enables stock [Chrome](https://www.google.com/chrome/) for Ubuntu to run on WSL 50046. I had mentioned doing this back in September #1121 ([message](https://github.com/Microsoft/BashOnWindows/issues/1121#issuecomment-249504662)), but Chrome was blocking on bigger problems at the time and can be compiled without udev support (notably for Android). There are still two blocking bugs outstanding, but it turns out they only manifest in debug builds (`DCHECK` fails). You can head on over the [github page](https://github.com/therealkenc/libudev-stub) to download and build if you feel lucky.

**Comment by therealkenc** [comment]
A binary package is also now ready. Non-tech steps for chrome on WSL:
```
sudo apt-get install ubuntu-desktop wget libxss1 libappindicator1 libindicator7
wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
sudo dpkg -i google-chrome-stable_current_amd64.deb
wget https://github.com/therealkenc/libudev-stub/releases/download/v0.9.0/libudev-stub-0.9.0-WSL.deb
sudo dpkg -i libudev-stub-0.9.0-WSL.deb
google-chrome --no-sandbox
```
There is (at least) one annoyance, being that chrome does not exit properly when closed. If this happens, hit `ctrl-c` a couple of times and then run `killall chrome nacl_helper` to clean up the process leftovers.

So, "workaround-available". Or something.

**Comment by mgiuffrida** [comment]
@therealkenc surprised you're building with nacl. I'd assume it would be easier to pare the build down.

What about `use_udev = false`, would that help? there's more stuff like this in [build/args/headless.gn](https://cs.chromium.org/chromium/src/build/args/headless.gn?type=cs&q=use_udev&l=39); see also `gn args <outdir> --list`

**Comment by iz0eyj** [comment]
@therealkenc strange error because X-Server is running:
google-chrome --no-sandbox
[20419:20419:0305/141529.840803:ERROR:browser_main_loop.cc(272)] Gtk: cannot open display:

I needed also "sudo apt-get install -f" for dependancies

**Comment by aseering** [comment]
@therealkenc -- ooh, really cool stuff!  Thanks for sharing!

**Comment by therealkenc** [comment]
I also got the dependency problem and hoped `libappindicator1` `libindicator7` in the dependencies above would help. I don't think that's a "WSL thing"; the problem pops up for Real Linux if you search google. If you happened to catch what `apt-get install -f` fixed, please post it and maybe we can get a definitive no-error install recipe. 

That is a strange error if X is running. You'd expect that error if the display variable was not set (export `DISPLAY=localhost:0`). If you just do an `xlogo` immediately after the fail does it pop up? Or `firefox`, which gets installed with `ubuntu-desktop` and also uses Gtk? The error here is from Gtk not Chrome proper.

If firefox runs but chrome does not, try adding `--disable-gpu` to the chrome commandline. That's a hail mary guess though, because it isn't getting far enough for GPU stuff to choke.

**Comment by therealkenc** [comment]
@mgiuffrida I am not building it at all now.  The stub is so people can run Google branded chrome out of the box. Yes `use_udev=0` is the way to go if you want to build chrome yourself, but you'll need this [patch](https://gist.github.com/therealkenc/b82a5cc4720b559fd751296d60f799b4) I posted earlier. Also note building chromium on WSL is no fun right now because of #1726, and `gn` has issues, I think because of #1353. GYP builds still work for the time being.

**Comment by therealkenc** [comment]
@mgiuffrida - I just realized you are the Google guy who posted earlier (welcome!). The `'udev=0'` build error with 'OS=="linux" is with `removable_storage_provider_linux.cc` and `serial_service_impl.cc`. It would be nice if those were squashed upstream, as well as for other platforms without `udev` like (apparently) [flatpak](https://github.com/mariospr/chromium-browser-flatpak/blob/master/0001-Fix-build-for-with-Duse_udev-0.patch). I posted a comment in crbug [284898](https://bugs.chromium.org/p/chromium/issues/detail?id=284898#c10).

**Comment by iz0eyj** [comment]
@therealkenc strange... Firefox starts with "export DISPLAY=0.0" but not Chrome.
This is the result:
```
federico@Bestia:~$ google-chrome --no-sandbox
[180:199:0307/132642.951848:ERROR:bus.cc(427)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: File o directory non esistente
Assertion 'pthread_mutex_unlock(&m->mutex) == 0' failed at pulsecore/mutex-posix.c:108, function pa_mutex_unlock(). Aborting.
Annullato (core dump creato)
```
Sorry but I don't remember what libraries fixed using -f because I forgot to write it... I know, I'm stupid!
I can try to unistall WSL and restart from scratch.
Would be possible to do it without install Ubuntu Dssktop? I use Xfce4 (multi window) mixed with the Win 10 Desktop.

Let me know if you need the test from scratch.

**Comment by iz0eyj** [comment]
P.S.: Great work @therealkenc

**Comment by therealkenc** [comment]
Right; a few tips (I'll put up some docs in the repo when I get a chance). If you have problems with pulsecore #486 ([message](https://github.com/Microsoft/BashOnWindows/issues/486#issuecomment-234930031)) just do a `apt-get purge pulseaudio` so chrome doesn't try to use it.

Not only is it _possible_ to run without Ubuntu desktop, I haven't even tried using it with Unity and Compiz. I tried (and failed) to explain this over in #637 ([message](https://github.com/Microsoft/BashOnWindows/issues/637#issuecomment-238461920) and [message](https://github.com/Microsoft/BashOnWindows/issues/637#issuecomment-258731061)). Installing the `ubuntu-desktop` package is just to get the libraries chrome expects on your system. Your mixed Xfce4 setup should fly.

Finally, you'll have a better (ie more stable) experience with the latest [mesa llvmpipe](https://www.mesa3d.org/llvmpipe.html). I built my own but there's a ppa [here](https://launchpad.net/~paulo-miguel-dias/+archive/ubuntu/mesa). You can see what you are running with `glxinfo -B`.

No need to track down the `install -f` further. It appears to be a common enough problem on Real Ubuntu and that's the standard fix. Thanks for trying it out!

**Comment by therealkenc** [comment]
Oh one more important tip I forgot to mention. Anecdotally, `apt-get` seems prone to blowing away libudev-stub's `libudev.so.1` symbolic link when installing or updating anything that depends on `udev`. It's just the one link, so you can fix it up manually, or re-install the libudev-stub .deb. You'll know when it happens because chrome just segfaults.

**Comment by iz0eyj** [comment]
```
federico@Bestia:~$ sudo apt-get purge pulseaudio
[sudo] password di federico:
Lettura elenco dei pacchetti... Fatto
Generazione albero delle dipendenze
Lettura informazioni sullo stato... Fatto
I seguenti pacchetti sono stati installati automaticamente e non sono più richiesti:
  libfftw3-single3 libpulsedsp libwebrtc-audio-processing-0 pulseaudio-utils rtkit
Usare "sudo apt autoremove" per rimuoverli.
I seguenti pacchetti saranno RIMOSSI:
  libcanberra-pulse* pulseaudio* pulseaudio-module-x11*
0 aggiornati, 0 installati, 3 da rimuovere e 0 non aggiornati.
Dopo quest'operazione, verranno liberati 4.136 kB di spazio su disco.
Continuare? [S/n] s
(Lettura del database... 145904 file e directory attualmente installati.)
Rimozione di libcanberra-pulse:amd64 (0.30-2.1ubuntu1)...
Rimozione di pulseaudio-module-x11 (1:8.0-0ubuntu3.2)...
Rimozione di pulseaudio (1:8.0-0ubuntu3.2)...
Eliminazione dei file di configurazione di pulseaudio (1:8.0-0ubuntu3.2)...
Elaborazione dei trigger per man-db (2.7.5-1)...
federico@Bestia:~$
federico@Bestia:~$ google-chrome --no-sandbox
[20854:20854:0308/142535.026220:ERROR:browser_main_loop.cc(272)] Gtk: cannot open display:
federico@Bestia:~$ export DISPLAY=localhost:0
federico@Bestia:~$ google-chrome --no-sandbox
[20866:20877:0308/142710.449012:ERROR:bus.cc(427)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: File o directory non esistente
Assertion 'pthread_mutex_unlock(&m->mutex) == 0' failed at pulsecore/mutex-posix.c:108, function pa_mutex_unlock(). Aborting.
Annullato (core dump creato)
federico@Bestia:~$

```
@therealkenc:
Interesting thing, X was running for Xfce but Chrome needs "export DISPLAY=localhost:0" to find it.

Yesterday I then uninstalled Ubuntu Desktop is possible that the problem lies; Tomorrow I will try to install it again and repeat the test, but I think the best thing is to start from scratch with a blank WSL and documenting each step.
The browser is something very much carrying the first that users install.
But the ideal would be to make it go with Xfce and maybe some additional library because for some install more than 1 GB of stuff then no use when employing the two mixed environments could be a problem.

**Comment by therealkenc** [comment]
Starting from scratch doesn't hurt but takes effort. Note I did _not_ have to uninstall pulseaudio, but for some reason the mutex bug doesn't manifest here. So I am a bit stumped.

Possibly a way forward is just to compile pulse yourself, with the one-line fix for `HAVE_PTHREAD_PRIO_INHERIT` [here](https://github.com/Microsoft/BashOnWindows/issues/486#issuecomment-235151066). There are reasonable build instructions [here](http://www.linuxfromscratch.org/blfs/view/svn/multimedia/pulseaudio.html). You don't need to actually run an audio server. We're just trying to make sure #1006 isn't tripped.

But to be clear I am _not_ doing this either. I haven't actually run audio since middle of last year. I am going to be really sad if after all this, it is silly audio that is blocking. Maybe I should just bundle up `libpulse.so` in the [libudev-stub](https://github.com/therealkenc/libudev-stub) installer. The whole point was not having to compile stuff.

No worries on your English. Il tuo inglese è migliore del mio italiano.

**Comment by lypanov** [comment]
This is getting especially interesting given the release of chrome-headless. Thanks everyone for your work on this!

**Comment by therealkenc** [comment]
Dev channel 60.0.3080.5 [Headless Chrome](https://developers.google.com/web/updates/2017/04/headless-chrome) runs out of the box with the [libudev-stub](https://github.com/therealkenc/libudev-stub) workaround. Below is a screencap of headfull chrome talking to headless chrome, both running simultaneously on WSL. The headless side is totally a "CLI development use case scenario". The other isn't. Or something.

Notable side remark from the Google developer in that link above, quote: "Since I'm on a Mac..." Something to think about.

![wsl-headless-chrome](https://cloud.githubusercontent.com/assets/7049700/25644474/3a30af68-2f5b-11e7-867c-7e336f0a1dd1.png)

**Comment by jgoux** [comment]
@therealkenc Thanks for your awesome work!
I followed your instructions and compiled pulseaudio by myself with the fix, but I still get the same error as @iz0eyj : `Assertion 'pthread_mutex_unlock(&m->mutex) == 0' failed at pulsecore/mutex-posix.c:108, function pa_mutex_unlock().` when trying to launch vscode.

**Comment by therealkenc** [comment]
If you are getting that assertion at line 108 of [`mutex-posix.c`](https://cgit.freedesktop.org/pulseaudio/pulseaudio/tree/src/pulsecore/mutex-posix.c), it probably means you don't have the `#undef HAVE_PTHREAD_PRIO_INHERIT` at around line 23 as described [here](https://github.com/Microsoft/BashOnWindows/issues/486#issuecomment-235151066) (or it wouldn't trip).

Since this is clearly blocking folks, I'll see if I can do a PulseAudio PPA like @aseering  [did](https://launchpad.net/~aseering/+archive/ubuntu/wsl-pulseaudio) for Trusty sometime this week. I have been loath to do one because, knowing my luck, #1006 will be squashed by Stephen about 24 hours after I put it up.

**Comment by therealkenc** [comment]
Okay I've done a new libpulse build for WSL Xenial. I jumped through the Launchpad PPA hoops, but I gather it takes time for the build bots to do their thing. It might or might not show up [here](https://launchpad.net/~therealkenc/+archive/ubuntu/wsl-pulseaudio) eventually. I also put up a `.deb` package in the libudev-stub github [release page here](https://github.com/therealkenc/libudev-stub/releases/tag/v0.9.0).

So the low tech steps, if you are feeling lucky, might be:
```
sudo apt-get install ubuntu-desktop       # and in general get firefox running with VcXsrv
sudo dpkg -i google-chrome-unstable_current_amd64.deb
sudo apt -f install                       # probably
sudo dpkg -i libpulse0_8.0-0ubuntu3.2ppa1_amd64.deb
sudo dpkg -i libudev-stub-0.9.0-WSL.deb   # doing this **last** is seriously important
google-chrome-unstable --no-sandbox
```
Again a heads up that the libudev-stub symbolic link will get blown away by `apt-get <almost anything>`, so if you get a segfault right out of the gate it probably means the link has been clobbered. Just reinstall the libudev-stub .deb to fix it up.

_Bonne chance._

**Comment by therealkenc** [comment]
The PulseAudio PPA [appeared](https://launchpad.net/~therealkenc/+archive/ubuntu/wsl-pulseaudio). So alternately for the PA bits:
```
sudo add-apt-repository ppa:therealkenc/wsl-pulseaudio
sudo apt-get update && sudo apt-get upgrade
sudo apt-get install -f   # might need to do this if you installed the .deb manually
sudo dpkg -i libudev-stub-0.9.0-WSL.deb  # because the symlink will be clobbered
```
Personally I would just use the `.deb`.

**Comment by jgoux** [comment]
And...it works! Thanks a ton for the libpulse deb!!!

I'm able to run Chrome now. 🏃 

I tried both code and code-insiders but still no luck.

This is the logs I have : 
![image](https://cloud.githubusercontent.com/assets/1443499/25780228/32d93c40-3324-11e7-9ce5-34c688a3f47f.png)

Did you succeed in launching vscode from WSL?

I'm using VcXsrv, which runs Chrome well.

**Comment by therealkenc** [comment]
Unfortunately vscode is a no-go at the moment. It is impractical to debug on WSL right now due to `gdb` limitations; and even on a good day without `gdb` limitations it would be a bear to debug down at this level because of all the libchromiumcontent threads and message passing between nodejs and the electron renderer process. The `strace` output doesn't have anything glaringly broken, so I can't even pinpoint places in the source add debug output lines.

I have been waiting for #1006 to flip and then perhaps try to find someone motivated enough on the vscode team to take a look and maybe make some educated guesses. It fails early in [vscode-loader](https://github.com/Microsoft/vscode-loader), which itself is a complicated package. Either vscode-loader is failing itself, or it tries to load some node native library that in turn hits a WSL gap not disturbed by Chrome proper. I also have an eye on #1910.

Atom gets a little further (at least it lights up), but can't open files. I don't use Atom so I didn't track the fail very far.

**Comment by jgoux** [comment]
Well, I suppose I'll have to wait! :p 

You already got us really far so thanks again for all the tools!

**Comment by therealkenc** [comment]
Per video drivers (saw your edit), what you want is mesa llvmpipe, which is also typical in a VM scenario.
```
sudo add-apt-repository ppa:paulo-miguel-dias/mesa
sudo apt-get update
sudo apt-get dist-upgrade
sudo apt-get install mesa-utils
sudo dpkg -i libudev-stub-0.9.0-WSL.deb  # because the symlink will be clobbered
glxinfo -B | grep OpenGL
```
If all goes well you should see:
```
OpenGL vendor string: VMware, Inc.
OpenGL renderer string: Gallium 0.4 on llvmpipe (LLVM 5.0, 256 bits)
OpenGL core profile version string: 3.3 (Core Profile) Mesa 17.2.0-devel - padoka PPA
...
```

**Comment by bmayen** [comment]
What would be the reason to run VSCode through WSL anyway? Chrome I understand for running tests against, but what's the benefit of running VSCode through it vs native?

**Comment by therealkenc** [comment]
NodeJS Linux Development. DotNet Linux development. C/C++ Linux development. 

You can in principal use Windows VSCode though WSL interop, but then you'll run headlong into DrvFS limitations (777 perms, symlinks, NTFS unfriendly filenames, files pinned by open handles). Chrome on WSL isn't all that useful (possibly notwithstanding headless), because all interaction goes through HTTP, which means it can just as well be running on Windows. VSCode talks to the filesystem, and there is the rub.

**Comment by aseering** [comment]
@therealkenc , "gdb":  For what it's worth, I've observed significant improvements in gdb's functionality in post-Creators Insider builds.  I'd be curious if vscode is now more debuggable.  (I've been traveling lately and haven't had a chance to update relevant tickets.)

**Comment by therealkenc** [comment]
@aseering - Unfortunately as of 16188 I still can't break with `ctrl-c` to see where vscode is stuck.
```
ken@DESKTOP-9BDQA3V:~/Devel/vscode/vscode$ gdb ../electron/out/D/electron
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.04) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
[...]
Reading symbols from ../electron/out/D/electron...done.
(gdb) set args .
(gdb) run
Starting program: /home/ken/Devel/vscode/electron/out/D/electron .
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffd9820700 (LWP 320)]
^C^C^C^C^C^C^C^C^C^C^C^C^C^C
```

**Comment by evil-shrike** [comment]
I followed [comment](#issuecomment-299681235) by @therealkenc, but  getting some error in /opt/google/chrome-unstable/libosmesa.so (the same was when I tried stable Chrome):
```
root@host:/mnt/c/temp/lx# google-chrome-unstable --no-sandbox --headless --remote-debugging-port=9222 https://chromium.org
UDEV STUB: Using libudev-stub 0.9.0
UDEV STUB: udev_new()
UDEV STUB: stubbing udev...
UDEV STUB: udev_monitor_new_from_netlink() name: udev
UDEV STUB: udev_monitor_filter_add_match_subsystem_devtype() subsystem: sound devtype: (null)
UDEV STUB: udev_monitor_filter_add_match_subsystem_devtype() subsystem: video4linux devtype: (null)
UDEV STUB: udev_monitor_enable_receiving()
UDEV STUB: udev_monitor_get_fd()
Segmentation fault (core dumped)
root@host:/mnt/c/temp/lx# [0525/171206.593849:ERROR:gl_implementation.cc(246)] Failed to load /opt/google/chrome-unstable/libosmesa.so: /opt/google/chrome-unstable/libosmesa.so: cannot open shared object file: No such file or directory
```
How can it be fixed?

**Comment by therealkenc** [comment]
@evil-shrike -- Sound a lot like [this](https://news.ycombinator.com/item?id=14240460). Try adding:
```
$ google-chrome-unstable --disable-gpu --no-sandbox --headless --remote-debugging-port=9222 https://chromium.org
```
You can also try installing `ppa:paulo-miguel-dias/mesa` mentioned [above](https://github.com/Microsoft/BashOnWindows/issues/648#issuecomment-299702548) and see if that is a big hammer fix. 

Are you running stock Ubuntu Xenial or some other distro? I ask because it should run out of the box like for jgoux so long as you have a slate that would also succeed on Real Linux. Anyway looks like you are close because you are getting the libudev-stub traces.

**Comment by evil-shrike** [comment]
@therealkenc thanks. adding `--disable-gpu` helped, the last error (libosmesa.so) has gone, other things stayed and Chrome just exists. Will try with mesa later. 
yes, I'm running stock Ubuntu for Creators Update (16)

**Comment by therealkenc** [comment]
Chrome now works out of the box _with sandbox_ on 16273 thanks to @benhillis work in #2242. There is a couple of caveats (there are always caveats). First, the `pthread_mutex_unlock(&m->mutex)` fail (which is #486) _will still trigger_ if you have pulse audio installed. Which you do, if you installed the whole ubuntu-desktop meta package. This had me confused [earlier](https://github.com/Microsoft/BashOnWindows/issues/648#issuecomment-284855564) as well (see the strikeout). There are two ways around. One is to push though it with the [PulseAudio 8 PPA](https://launchpad.net/~therealkenc/+archive/ubuntu/wsl-pulseaudio). The other is to purge pulse from your system. I'd recommend the latter.

The other caveat, which is of the soft variety, is that chrome would really rather be living in an environment with a functioning dbus and opengl (ie mesa llvmpipe). Putting it all together, something like the following should get you there:

```
sudo update && sudo apt upgrade
sudo apt install ubuntu-desktop wget mesa-utils
sudo apt purge pulseaudio libpulse0 && sudo apt autoremove
wget https://dl.google.com/linux/direct/google-chrome-unstable_current_amd64.deb
sudo dpkg -i google-chrome-unstable_current_amd64.deb
sudo apt -f install                       # probably
export DISPLAY=localhost:0
export NO_AT_BRIDGE=1
exec dbus-run-session -- bash
sudo service dbus start
google-chrome-unstable           # look mom, no flags
```
And there you have it.

**Comment by iz0eyj** [comment]
GREAT!!! it works fine!!!
A peoblem: xfce4 needs pulseaudio :(
So, if I install Chrome... bye bye desktop integration with xfce :(

**Comment by iz0eyj** [comment]
Hey @therealkenc this is my preferred browser, Vivaldi, that runs out of the box without any hack.
Many thanks to tou and to @benhillis 

![desktop](https://user-images.githubusercontent.com/6331284/29672127-a46204c4-88eb-11e7-836d-09bef72e917b.jpg)

**Comment by benhillis** [comment]
@therealkenc - Unfortunately I didn't have enough time to get the priority futex work done for this release.  I'm hoping to knock this one out very early for the next release.

**Comment by davidmaxwaterman** [comment]
FYI, I've been using WSL since late last year. I would normally wipe off Windows and install Ubuntu, but WSL has been working for me since then. However, I am now trying to write some tests for my Google Polymer based web apps, and the tools use Web Component Tester, and so I hit this problem where it can't launch the browser to run my tests. It does seem like there is a solution trying to find it's way out, but the whole 'no GUI' stance makes me think it'll not be supported for some time, even if it is made to work soon (which seems likely). As such, I'm think I'm going to switch back to Ubuntu, which is a bit of a shame, tbh....WSL is kind of awesome really.

**Comment by jehon** [comment]
I have the impression that google chrome works if you:
- install a XWindows server on your windows box (I use VcXsrv X SErver 1.19.6.0 - it include GxLib 1.3 (? not sure of the lib ?) wich is needed by chrome)
- export DISPLAY=:0

But since that's the conclusion of a long work, with lots of trials, I would rather hope that someone else could confirm that...

**Comment by bitcrazed** [comment]
@davidmaxwaterman - I know that @TheLarkInn was looking into how one could launch a browser from within WSL.

**Comment by therealkenc** [comment]
> how one could launch a browser from within WSL.

[this](https://github.com/Microsoft/WSL/issues/648#issuecomment-262803652). There is nothing special about launching a Windows browser (Chrome/Edge/Firefox) versus any other Windows application (Notepad). Native Linux Chrome has worked since 16273 and this issue is fixedinsiderupdate. There's no known issues filed in github related to Chrome, with the nonblocking exceptions of #1006 and #1353.

**Comment by davidmaxwaterman** [comment]
Thanks but I wrote that 5 months ago and have long since given up with WSL and gone back to Ubuntu. I can quite believe things have changed since then.

**Comment by rsshilli** [comment]
This works great for me.  I installed [VcXsrv ](https://sourceforge.net/projects/vcxsrv/)and then followed @therealkenc 's comment from Aug 24, 2017. Thank you!

**Comment by yxy1996** [comment]
I came across with this situation When I launched roboware. I tried to use the latter method, but it will purge so many packages relative to ros-kinetic. I tried it eventually and my roboware worked magically, however, `roscd` couldn't work any more.

**Comment by illyaMs** [comment]
Combining the tips from @therealkenc (https://github.com/Microsoft/WSL/issues/648#issuecomment-324562271) and @jehon (https://github.com/Microsoft/WSL/issues/648#issuecomment-372794217) worked for me as well. Had issue running automated test based on Chromium from win10/WSL.

**Comment by darrenm95** [comment]
I still get an issue when trying to run google chrome on windows subsystem for linux. Using ubuntu 16.04. Has anyone got any ideas why I am getting the followings errors?


[28834:28834:0206/151626.871150:ERROR:browser_switcher_service.cc(238)] XXX Init()
[28918:32:0206/151627.109653:ERROR:platform_shared_memory_region_posix.cc(52)] Descriptor access mode (2) differs from expected (0)
[28943:50:0206/151627.224939:ERROR:platform_shared_memory_region_posix.cc(52)] Descriptor access mode (2) differs from expected (0)
[28929:36:0206/151627.235707:ERROR:platform_shared_memory_region_posix.cc(52)] Descriptor access mode (0) differs from expected (2)
[28961:65:0206/151627.236391:ERROR:platform_shared_memory_region_posix.cc(52)] Descriptor access mode (2) differs from expected (0)
[28868:28868:0206/151627.432404:ERROR:gl_surface_glx.cc(459)] GLX 1.3 or later is required.
[28868:28868:0206/151627.432430:ERROR:gl_initializer_x11.cc(148)] GLSurfaceGLX::InitializeOneOff failed.
[28868:28868:0206/151627.438697:ERROR:viz_main_impl.cc(159)] Exiting GPU process due to errors during initialization
[28834:28834:0206/151628.837524:ERROR:CONSOLE(1)] "Uncaught SyntaxError: Unexpected token '<'", source: chrome://welcome/welcome_app.js (1)
[28894:13:0206/151628.841397:ERROR:platform_shared_memory_region_posix.cc(52)] Descriptor access mode (0) differs from expected (2)
[28834:28853:0206/151629.981332:FATAL:gpu_data_manager_impl_private.cc(439)] GPU process isn't usable. Goodbye.
Trace/breakpoint trap (core dumped)

**Comment by ambrt** [comment]
Have same thing .  I came to flags below. It flashes the website correctly (in VcXsrv)
then crashes
`sudo DISPLAY=:0 google-chrome   --disable-features=VizDisplayCompositor --no-sandbox    --disable-setuid-sandbox  --disable-dev-shm-usage     --unlimited-storage     --ignore-certificate-errors   --enable-font-antialiasing      --font-render-hinting=medium          http://example.com`


The error stack is this
```
[30371:30371:0319/144325.480839:ERROR:edid_parser.cc(102)] Too short EDID data: manufacturer id
[30371:30398:0319/144325.630878:ERROR:bus.cc(393)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[30371:30398:0319/144325.650220:ERROR:bus.cc(393)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[30371:30398:0319/144325.650262:ERROR:bus.cc(393)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[30371:30451:0319/144325.809533:ERROR:object_proxy.cc(632)] Failed to call method: org.freedesktop.DBus.Properties.Get: object_path= /org/freedesktop/UPower: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.UPower was not provided by any .service files
[30371:30451:0319/144325.810376:ERROR:object_proxy.cc(632)] Failed to call method: org.freedesktop.UPower.GetDisplayDevice: object_path= /org/freedesktop/UPower: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.UPower was not provided by any .service files
[30371:30451:0319/144325.811266:ERROR:object_proxy.cc(632)] Failed to call method: org.freedesktop.UPower.EnumerateDevices: object_path= /org/freedesktop/UPower: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.UPower was not provided by any .service files
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
[30396:30396:0319/144325.943947:ERROR:viz_main_impl.cc(161)] Exiting GPU process due to errors during initialization
[30371:30383:0319/144326.480351:ERROR:browser_gpu_channel_host_factory.cc(138)] Failed to launch GPU process.
Xlib:  extension "MIT-SHM" missing on display ":0".
```

The network seems to work because if i will run with `sudo DISPLAY=:1 xvfb-run (...) --dump-dom  --headless  http://example.com`
It dumps dom correctly then crashes with

```
[0319/144741.169122:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 1 time(s)
[0319/144741.313374:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 2 time(s)
[0319/144741.462192:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 3 time(s)
[0319/144741.611385:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 4 time(s)
[0319/144741.767465:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 5 time(s)
[0319/144741.926318:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 6 time(s)
[0319/144741.927170:ERROR:browser_gpu_channel_host_factory.cc(138)] Failed to launch GPU process.
```

**Comment by G-Rath** [comment]
@therealkenc I've followed your instructions in your [original comment](https://github.com/microsoft/WSL/issues/648#issuecomment-324562271), but chrome seems to having some issues.

Here are some details:

Windows 10 Pro
Version: 1909
OS Build: 18363.720

VcXsrv X Server: 1.20.6.0 (12 Jan 2020)
Google Chrome 80.0.3987.149

I can run both `google-chrome-stable` & `google-chrome-unstable` if I use `sudo` & `--no-sandbox`, but `--headless` crashes:

```
g-rath@DESKTOP-MUGH739:/c/Users/G-Rath$ sudo google-chrome-stable --no-sandbox --headless https://google.com
[0320/095942.195820:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 1 time(s)
[0320/095942.337443:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 2 time(s)
[0320/095942.473931:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 3 time(s)
[0320/095942.610071:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 4 time(s)
[0320/095942.748664:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 5 time(s)
[0320/095942.882727:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 6 time(s)
[0320/095942.978725:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 7 time(s)
[0320/095943.069493:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 8 time(s)
[0320/095943.157845:WARNING:gpu_process_host.cc(1165)] The GPU process has crashed 9 time(s)
[0320/095943.157879:FATAL:gpu_data_manager_impl_private.cc(1034)] The display compositor is frequently crashing. Goodbye.
Failed to generate minidump.Illegal instruction (core dumped)
```

This is very annoying because headless is what I want, so I can run feature tests 😬 

Heres the output for trying to run `google-chrome-stable`:

```
g-rath@DESKTOP-MUGH739:/c/Users/G-Rath$ /usr/bin/google-chrome-stable
Failed to move to new namespace: PID namespaces supported, Network namespace supported, but failed: errno = Permission denied
Trace/breakpoint trap (core dumped)
g-rath@DESKTOP-MUGH739:/c/Users/G-Rath$ stat /usr/bin/google-chrome-stable
  File: /usr/bin/google-chrome-stable -> /opt/google/chrome/google-chrome
  Size: 32              Blocks: 0          IO Block: 512    symbolic link
Device: 2h/2d   Inode: 844424932927321  Links: 1
Access: (0777/lrwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2020-03-19 17:46:48.000000000 +1300
Modify: 2020-03-17 10:50:01.000000000 +1300
Change: 2020-03-19 17:46:55.208091200 +1300
 Birth: -
```

If I then run it with `--no-sandbox`, Chrome opens and then crashes with a lot of permission errors.

For some reason it seems that Chrome is creating its cache (and I assume other files) in a way that means they end up being owned by root?

Let me know if you'd like me to create a new issue, or have any questions :)

**Comment by ambrt** [comment]
The comment above about not working solution must be related to chrome version. The newest version 80.* is failing me also.

I used version 77 that i happened to have installed on wsl some time ago and it manages to open non https website, via VcXsrv.
`sudo DISPLAY=:0 /home/user/node_modules/puppeteer/.local-chromium/linux-674921/chrome-linux/chrome   --disable-features=VizDisplayCompositor --no-sandbox    --disable-setuid-sandbox  --disable-dev-shm-usage     --unlimited-storage     --ignore-certificate-errors     --disable-gpu     --enable-font-antialiasing      --font-render-hinting=medium      --use-gl=wgl       --disable-software-rasterizer      http://example.com`

It still opens https sites as a blank page.

**Comment by larzuk91** [comment]
I downloaded the latest version today (81.0.4044) and I got this error as well. I don't know why this ticket is closed =/

**Comment by savethebyte** [comment]
I have used chrome 83 and 84, and I am still recieving the error in WSL.

`[0709/221720.544454:ERROR:platform_shared_memory_region_posix.cc(46)] Descriptor access mode (0) differs from expected (2)
[0709/221720.548263:WARNING:crash_handler_host_linux.cc(366)] Could not translate tid - assuming crashing thread is thread group leader; syscall_supported=0
[0709/221720.675739:WARNING:crash_handler_host_linux.cc(366)] Could not translate tid - assuming crashing thread is thread group leader; syscall_supported=0
--2020-07-09 22:17:20--  https://clients2.google.com/cr/report
Resolving clients2.google.com (clients2.google.com)... 172.217.6.110, 2607:f8b0:4009:813::200e
Connecting to clients2.google.com (clients2.google.com)|172.217.6.110|:443... connected.
[1594351040.818][DEBUG]: DevTools WebSocket Event: Inspector.targetCrashed 5A70B9C03C3188FDCA221C0B94718B5F {

}
[1594351040.818][INFO]: Done waiting for pending navigations. Status: tab crashed
HTTP request sent, awaiting response... [1594351040.868][INFO]: [c6f38322f1211be529223d6f0a3aafb7] RESPONSE ClickElement ERROR unknown error: session deleted because of page crash
from tab crashed
  (Session info: headless chrome=83.0.4103.116)
[1594351040.868][DEBUG]: Log type 'driver' lost 0 entries on destruction
[1594351040.868][DEBUG]: Log type 'browser' lost 3 entries on destruction
200 OK
Length: unspecified [text/plain]
Saving to: ‘/dev/fd/4’

Crash dump id: aa6abc501aeb718e

     0K                                                         744K=0s`

**Comment by kohheepeace** [comment]
I followed this docs https://docs.microsoft.com/en-us/windows/wsl/tutorials/gui-apps

And I got these errors

```sh
$ google-chrome
[2014:2014:1112/172745.466766:ERROR:browser_dm_token_storage_linux.cc(94)] Error: /etc/machine-id contains 0 characters (32 were expected).
[1112/172745.479719:ERROR:file_io_posix.cc(144)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory (2)
[1112/172745.479781:ERROR:file_io_posix.cc(144)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: No such file or directory (2)
Trace/breakpoint trap
```

**Comment by ricklambrechts** [comment]
For the people that are expierencing issues after using: https://docs.microsoft.com/en-us/windows/wsl/tutorials/gui-apps

Check the wsl version with:
```
wsl --list --verbose
```

If the wsl distro is version 1 then convert the distro to version 2 with:
Check this documentation: https://docs.microsoft.com/en-us/windows/wsl/basic-commands#set-wsl-version-to-1-or-2
```
wsl --set-version <name> 2
```

**Comment by mochammadrafi** [comment]
Any solution?

**Comment by simonleary-umass-edu** [comment]
A simple `wsl --update` fixed this issue for me. I'm not sure what version I had because `wsl --version` apparently wasn't a valid command back then. I'm now using 1.2.5.0

**Comment by gelodefaultbrain** [comment]
Hi, what did you guys do? I just did 
```
gelswsl@LAPTOP-MCLLQCCH:~$ google-chrome
Killed
```

what is happening?

**Comment by gelodefaultbrain** [comment]
> Hi, what did you guys do? I just did
> 
> ```
> gelswsl@LAPTOP-MCLLQCCH:~$ google-chrome
> Killed
> ```
> 
> what is happening?

Hi Had to update wsl as well as mentioned [here](https://github.com/microsoft/WSL/issues/648#issuecomment-1577568742). Well with that it caused my running application to restart as well 😢
